lib/src/ui/pages/settings/settings_notifier.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:allthenews/src/domain/appinfo/app_info_repository.dart';
+import 'package:allthenews/src/domain/common/function.dart';
+import 'package:allthenews/src/domain/settings/popular_news_criterion.dart';
+import 'package:allthenews/src/domain/settings/settings_repository.dart';
+import 'package:allthenews/src/domain/settings/app_theme.dart';
+import 'package:allthenews/src/ui/common/theme/theme_notifier.dart';
+import 'package:flutter/material.dart';
+
+class SettingsNotifier extends ChangeNotifier {
+  final AppInfoRepository _appInfoRepository;
+  final SettingsRepository _settingsRepository;
+
+  SettingsNotifier(this._settingsRepository, this._appInfoRepository);
+
+  SettingsViewState _viewState = SettingsViewState(
+    isLoading: true,
+    appVersion: null,
+    isDarkModeEnabled: false,
```

Comment:
> I recommend defaulting to System for three reasons. Firstly, most users who care about light and dark modes already chose their preferred mode using the system. The users’ default preference is already defined. Secondly, for users who have the option and don’t care, every other part of the system uses the system value (generally, light mode). Thirdly, it acts like every other app when running on older versions of the OS that don’t support light/dark themes.

https://medium.com/@pmutisya/dark-mode-in-flutter-3742062f9f59

Reply:
Zgadzam się, ale... taki był projekt przecież :D na switchu są dwa stany, ciemny-jasny - musiałby być trzeci stan jako systemowy, żeby to uwzględnić, bo domyślnie nie dam systemowego, a jak zmieni to już taki na jaki zmienił, bo użytkownik nawet sobie nie będzie zdawał z tego sprawy, że tak jest i po zmianie nawet nie będzie mógł wrócić do systemowego :P

Reply:
no ten design zrobiliśmy z Pawłem w ten sposób, że rzeczywiście ciężko tam systemowy dark mode wcisnąć :P dropdown byłby lepszy, ale teraz to już chuj